authzed/api/v1/permission_service.proto

```diff
@@ -122,24 +122,42 @@ message Consistency {
 // RelationshipFilter is a collection of filters which when applied to a
 // relationship will return relationships that have exactly matching fields.
 //
-// resource_type is required. All other fields are optional and if left
-// unspecified will not filter relationships.
+// All fields are optional and if left unspecified will not filter relationships,
+// but at least one field must be specified.
+//
+// NOTE: The performance of the API will be affected by the selection of fields
+// on which to filter. If a field is not indexed, the performance of the API
+// can be significantly slower.
 message RelationshipFilter {
+  // resource_type is the *optional* resource type of the relationship.
+  // NOTE: It is not prefixed with "optional_" for legacy compatibility.
```

Comment:
it's the number `1` that provides the compatibility, right?  are you concerned about the generated client APIs changing?

Reply:
Yeah, I'm concerned it'll break existing client code

Reply:
It shouldn't break existing client code, it would break when someone updates their version and finds codegen changed the field name

Reply:
That's what I meant, but it is still a "breaking" change in the sense that moving a minor version of the client library could break existing callers. We'd either have to major version all the client libraries or just not change it